metadata/definition/definition.go

```diff
@@ -34,6 +42,36 @@ type ServiceDefinition struct {
 	Types         []TypeDefinition
 }
 
+func (def ServiceDefinition) ToBytes() ([]byte, error) {
+	return json.Marshal(def)
+}
+
+func (def ServiceDefinition) String() string {
+	var methodStr strings.Builder
+	for _, m := range def.Methods {
+		var paramType strings.Builder
+		for _, p := range m.ParameterTypes {
+			paramType.WriteString(fmt.Sprintf("{type:%v}", p))
```

Comment:
fmt.Fprintf(paramType, "{type:%v}", p) ?

Reply:
？？

Reply:
> if you wanna got its type, u should use `%T`.

type is just the variable's name. Not reflect.type meaning.